scheduler/src/cook/kubernetes/api.clj

```diff
@@ -1836,6 +1837,48 @@
               (log/info "In" compute-cluster-name "compute cluster, pod" pod-name "was already deleted")
               (throw e))))))))
 
+(defn delete-collect-results-finalizer
+  "Delete the finalizer that collects the results of a pod completion state once the pod has a
+  deletion timestamp"
+  [{:keys [compute-cluster-name api-client name]} pod-name ^V1Pod pod]
+  (let [^V1ObjectMeta pod-metadata (some-> pod .getMetadata)
+        pod-deletion-timestamp (some-> pod .getMetadata .getDeletionTimestamp)
+        ^List finalizers (some-> pod-metadata .getFinalizers)]
+    (when (and finalizers (> (.size finalizers) 1))
+      ; Finalizers are stored in a list and deleted by index. If there's more than one,
+      ; cook can race with whatever deletes the other one, or even delete both (see race discussed below)
+      (log/error "In compute-cluster" name ", deleting finalizer for pod " pod-name
+                 " Cook's deletion of more than one finalizer on a pod is racy and not supported. "
+                 "Finalizers are" finalizers))
+    (when (and pod-deletion-timestamp (some #(= FinalizerHelper/collectResultsFinalizer %) finalizers))
+      (log/info "In compute-cluster" name ", deleting finalizer for pod" pod-name)
```

Comment:
Can we make the start of these two logs consistent too?

Reply:
Ok.